server/middlewares/validate.ts

```diff
@@ -12,7 +12,8 @@ export default function validate<T extends z.ZodType<BaseReq>>(schema: T) {
       ctx.input = schema.parse(ctx.request);
     } catch (err) {
       const { path, message } = err.issues[0];
-      const [prefix = "ValidationError"] = path;
+      const prefix =
+        path.length > 0 ? path[path.length - 1] : "ValidationError";
```

Comment:
Isn't `ValidationError` as part of the message a bit redundant here? Perhaps better to just pass `message` in that case…